pkg/reconciler/taskrun/taskrun.go

```diff
@@ -228,27 +230,15 @@ func (c *Reconciler) checkPodFailed(tr *v1beta1.TaskRun) (bool, v1beta1.TaskRunR
 	return false, "", ""
 }
 
-func (c *Reconciler) durationAndCountMetrics(ctx context.Context, tr *v1beta1.TaskRun) {
+func (c *Reconciler) durationAndCountMetrics(ctx context.Context, tr *v1beta1.TaskRun, beforeCondition *apis.Condition) {
 	logger := logging.FromContext(ctx)
 	if tr.IsDone() {
-		newTr, err := c.taskRunLister.TaskRuns(tr.Namespace).Get(tr.Name)
-		if err != nil && !k8serrors.IsNotFound(err) {
-			logger.Errorf("Error getting TaskRun %s when updating metrics: %w", tr.Name, err)
-			return
-		} else if k8serrors.IsNotFound(err) || newTr == nil {
-			logger.Debugf("TaskRun %s not found when updating metrics: %w", tr.Name, err)
-			return
-		}
-
-		before := newTr.Status.GetCondition(apis.ConditionSucceeded)
-		go func(metrics *taskrunmetrics.Recorder) {
-			if err := metrics.DurationAndCount(ctx, tr, before); err != nil {
-				logger.Warnf("Failed to log the metrics : %v", err)
-			}
-			if err := metrics.CloudEvents(ctx, tr); err != nil {
-				logger.Warnf("Failed to log the metrics : %v", err)
-			}
-		}(c.metrics)
+		if err := c.metrics.DurationAndCount(ctx, tr, beforeCondition); err != nil {
+			logger.Warnf("Failed to log the metrics : %v", err)
+		}
+		if err := c.metrics.CloudEvents(ctx, tr); err != nil {
+			logger.Warnf("Failed to log the metrics : %v", err)
+		}
```

Comment:
these 2 warn logs the same message (though err is different). This may be confusing for debugging. Could we change them into different msgs?

Reply:
makes sense, done!